saleor/plugins/webhook/tasks.py

```diff
@@ -160,15 +190,205 @@ def send_webhook_using_google_cloud_pubsub(
+            webhook.target_url,
+            delivery.event_type,
+            delivery.id,
+        )
+    except ValueError as e:
+        response = WebhookResponse(content=str(e), status=EventDeliveryStatus.FAILED)
+        attempt_update(attempt, response)
+        delivery_update(delivery=delivery, status=EventDeliveryStatus.FAILED)
+    clear_successful_delivery(delivery)
+
+
+def send_webhook_request_sync(app_name, delivery):
+    event_payload = delivery.payload
+    data = event_payload.payload
+    webhook = delivery.webhook
+    parts = urlparse(webhook.target_url)
+    domain = Site.objects.get_current().domain
+    message = data.encode("utf-8")
+    signature = signature_for_payload(message, webhook.secret_key)
+
+    response = WebhookResponse(content="")
+    response_data = None
+    if parts.scheme.lower() in [WebhookSchemes.HTTP, WebhookSchemes.HTTPS]:
+        logger.debug(
+            "[Webhook] Sending payload to %r for event %r.",
+            webhook.target_url,
+            delivery.event_type,
+        )
+        attempt = create_attempt(delivery=delivery, task_id=None)
+        try:
+            with webhooks_opentracing_trace(
+                delivery.event_type, domain, sync=True, app_name=app_name
+            ):
+                response = send_webhook_using_http(
+                    webhook.target_url,
+                    message,
+                    domain,
+                    signature,
+                    delivery.event_type,
+                    timeout=WEBHOOK_SYNC_TIMEOUT,
+                )
+                response_data = json.loads(response.content)
+        except RequestException as e:
+            logger.warning(
+                "[Webhook] Failed request to %r: %r. "
+                "ID of failed DeliveryAttempt: %r . ",
+                webhook.target_url,
+                e,
+                attempt.id,
+            )
+            response.status = EventDeliveryStatus.FAILED
+            response.content = e.response
+
+        except JSONDecodeError as e:
+            logger.warning(
+                "[Webhook] Failed parsing JSON response from %r: %r."
+                "ID of failed DeliveryAttempt: %r . ",
+                webhook.target_url,
+                e,
+                attempt.id,
+            )
+            response.status = EventDeliveryStatus.FAILED
+            response.content = e.msg
+        else:
+            logger.debug(
+                "[Webhook] Success response from %r."
+                "Succesfull DeliveryAttempt id: %r",
+                webhook.target_url,
+                attempt.id,
+            )
+
+        attempt_update(attempt, response)
+    else:
+        delivery_update(delivery, EventDeliveryStatus.FAILED)
+        raise ValueError("Unknown webhook scheme: %r" % (parts.scheme,))
+    delivery_update(delivery, response.status)
+    clear_successful_delivery(delivery)
+    return response_data
+
+
+# DEPRECATED
+# to be removed in task: #1q2x7xw
```

Comment:
I don't think that adding task id is a good way for this. Maybe we should add a number of version or time period? Something more "public"

Reply:
This will be changed right after releasing to cloud, the sooner the better. We just need to wait for existing Celery tasks to execute and we are free to drop this code.